students/A_Zach/lesson03/mailroom1.py

```diff
@@ -0,0 +1,129 @@
+import sys
+
+#Functions
+#Show list of names
+def show_list(inp):
+    #print names of donors
+    for name in names:
+        print(name)
+    return
+    
+#Add a new name to the list
+def search_name(inp):
+    response_donation = input("Please enter an amount to donate >")
+    if response_donation.lower == 'quit':
+        return
+    i = 0
+    #iterate through the database until the requested name is found
+    while i < len(names):
+        #Add the donation to the name provided
+        if names[i] == inp.title():
+            donations[i].append(float(response_donation))        
+            return
+        #Increase iteration if name has not been reached
+        i += 1
+    names.append(inp.title())
+    donations.append([float(response_donation)])
+    return response_donation
+
+#make the program quit
+def quit_prog(p):
+    if p.lower() == 'quit':
+        print("Thank you. Goodbye!")
+        sys.exit()
+
+#generate an email1
+def email(inp):
+    #find the index of n in the database
+    i = names.index(inp.title())
+    #format email. format the donation amount (d[i])
+    #into a floating point with 2 decimal places.
+    print(f"\n\nDear {inp},\n\n\tWe appreciate your generous donations totalling ${sum(donations[i]):.2f}.\n\nThank you,\nAndrew\n\n")
+    
+   
+    
+def send_thank_you():
+    inp = input("Who whould you like to thank?>")
+    if inp.lower == 'quit':
+        return
+    #show the list of names using show_list function
+    while inp.title() == 'List':
+        show_list(inp)
+        inp = input("Who whould you like to thank?>")
+        if inp.lower == 'quit':
+            return
+    
+    #Call the search_name function
+    donation = search_name(inp)
+    
+    #call the email function
+    email(inp)
+
+#creat report inputs    
+def report_results():
+    tots = []
+    aves = []
+    nums = []
+    for donation_list in donations:        
+        tots.append(sum(donation_list))
+        aves.append(sum(donation_list)/len(donation_list))
+        nums.append(len(donation_list))
+    return (tots, aves, nums)
```

Comment:
you can generate a list of tuples here -> `[(name, avg, total, num_of_donations),]` and use that with sort directly